lib/thinking_sphinx/active_record/source_joins.rb

```diff
@@ -54,4 +54,11 @@ def joins
       joins
     end
   end
+
+  # Use "first" here instead of a more intuitive flatten because flatten
+  # will also ask each column to become an Array and that will start
+  # to retrieve data.
+  def source_query_columns
+    @source.fields.select { |field| field.source_type == :query }.map(&:columns).map(&:first)
```

Comment:
A couple of house style things to note (just to keep the code consistent within TS):

* I prefer references to private instance methods representing the instance variables, rather than direct references to the instance variables. It can make refactoring later easier. So, `@source` should just be `source`.
* I prefer `collect` over `map` for the linguistic similarity of collect/select/detect/inject.

And then, I'm wondering if it's worth splitting this out into two methods - one that returns the query fields (i.e. the first half of what's in this original method), and then a second method with a predicate (say, `column_included_in_queries?(column)`) which takes in a column, and takes the second half of your original method, plus the `include?` check? That way, it's a slightly closer match to the `column_exists?` method.